### ISSUE TITLE
o:validateBean method="validateCopy" does not validate inputs in composite

### ISSUE THREAD

**Issue by BalusC** [omnifaces/omnifaces#155]
http://stackoverflow.com/q/31700829/157882

> I tracked down to the omnifaces source and got to the source point below, in the o:validateBean:

>     ValueReference valueReference = getValueReference(context.getELContext(), valueExpression);
>     if (valueReference.getBase().equals(base)) {
>         operation.run((EditableValueHolder) component, valueReference);
>     }

> In a case where I use a simple input text, the valueReference().getBase() returns my to be validated bean. In a case where I use a composite component, the valueReference().getBase() returns a reference to CompositeComponentAttributesELResolver.ExpressionEvalMap.

**Comment by BalusC** [comment]
Can't reproduce it with code provided on showcase http://showcase.omnifaces.org/validators/validateBean and below MCVE

```
<h:form>
	<cc:input value="#{validateClassLevelBean.product.number1}" />
	<cc:input value="#{validateClassLevelBean.product.number2}" />
    <h:commandButton value="submit" action="#{validateClassLevelBean.action}">
        <f:ajax execute="@form" render="@form"/>
    </h:commandButton>
    <h:messages />
	<o:validateBean value="#{validateClassLevelBean.product}" validationGroups="com.example.ProductGroup" method="validateCopy" />
</h:form>
```

```
<cc:interface>
	<cc:attribute name="value" />
    <cc:editableValueHolder name="input" targets="input" />
</cc:interface>
<cc:implementation>
	<h:inputText id="input" value="#{cc.attrs.value}" />
</cc:implementation>
```

Can you please provide information what's missing as compared to your case?

**Comment by BalusC** [comment]
Any feedback?

**Comment by BalusC** [comment]
Closing off as invalid.

**Comment by fwelland** [comment]
So this still doesn't work for me.   I think I follow your example exactly.   
My Environment:   WildFly 10.  JDK 1.8_77 (linux fwiw);  oh and:  'org.omnifaces:omnifaces:2.2'
The form:
````
        <h:form >
            <br/>
            <h:panelGroup layout="block" >
                Playing with o:validateBean...
            </h:panelGroup>
            <br/>
            <h:panelGrid frame="box" columns="2" id="ohMy" >

                <h:outputLabel value="First Name" for="fname"/>
                <h:inputText  id="fname" value="#{vladTheController.searchCritter.firstName}"/>

                <h:outputLabel value="Last Name" />
                <ez:labledInput value="#{vladTheController.searchCritter.lastName}" />                               
                
                <h:outputLabel value="ID" for="eyeD"/>
                <h:inputText  id="eyeD" value="#{vladTheController.searchCritter.id}"/>

                <h:outputLabel value="Tag Line" for="tline"/>
                <h:inputText  id="tline" value="#{vladTheController.searchCritter.tagLine}"/>

                <h:panelGroup />
                <h:commandButton value="submit" action="#{vladTheController.submit()}">

                </h:commandButton>

                <h:messages />
                <h:panelGroup />

            </h:panelGrid>
            <o:validateBean value="#{vladTheController.searchCritter}" validationGroups="fhw.SearchCritterVladGroup" method="validateCopy" />
        </h:form>
````

The Composite: 
````
<ui:component xmlns="http://www.w3.org/1999/xhtml"
        xmlns:cc="http://xmlns.jcp.org/jsf/composite"
        xmlns:ui="http://xmlns.jcp.org/jsf/facelets"
        xmlns:h="http://xmlns.jcp.org/jsf/html">

<cc:interface>
    <cc:attribute name="value" />
    <cc:editableValueHolder name="input" targets="input" />
</cc:interface>
<cc:implementation>
    <h:inputText id="input" value="#{cc.attrs.value}" />
</cc:implementation>    

</ui:component> 
````


Some SOUT stuff: 
````
16:42:56,471 INFO  [stdout] (default task-120) the copy is: SearchCritter(firstName=fred, lastName=null, id=123, tagLine=cheeze)
16:42:56,471 INFO  [stdout] (default task-120) the critter is: SearchCritter(firstName=fred, lastName=aaa, id=123, tagLine=cheeze)
````

The first output line comes from the validator that is working on the copy, or that is what I think is supposed to happen, correct? 

The second output line comes from the controller's submit() method.   

Notice lastName is not the same between the copy and 'the critter'.      As you see, of the 4 attrs in the form, lastName is the only CC.

**Comment by BalusC** [comment]
Finally reproduced and fixed in #240.
